### DISCUSSION TITLE
Why do `T` and `U` must have the same alignment in `try_cast_box`?

### DISCUSSION THREAD

**Discussion by gymore-io** [Lokathor/bytemuck#62] [Q&A] [answered]
I do not understand why casting from `T` to `U` were `align_of::<U>()` is less than `align_of::<T>()` isn't valid. Is there some requirements on `Box<T>` that I'm not aware of?

**Comment by Lokathor** [comment, marked as answer]
When the Box drops it will de-allocate. The requirements of the de-allocation is that it's de-allocated with the same size and alignment as was used to make the allocation.

**Reply by gymore-io** [reply]
Oh, I see. Thanks a bunch!
